src/coreclr/jit/hwintrinsicarm64.cpp

```diff
@@ -1769,9 +1769,9 @@ GenTree* Compiler::impSpecialIntrinsic(NamedIntrinsic        intrinsic,
         {
             assert(sig->numArgs == 2);
 
-            if (varTypeIsLong(simdBaseType))
+            if (varTypeIsLong(simdBaseType) && (intrinsic == NI_Vector64_op_Multiply))
             {
-                // TODO-ARM64-CQ: We should support long/ulong multiplication.
+                // long/ulong multiplication needs no special handling for SIMD8
```

Comment:
This is because `move to gpr, multiply, move to simd` is already as efficient as we can get?

Reply:
Mostly wondering why we'd convert `Vector128<long>` to an explicit IR sequence of `GetElement * GetElement` but we wouldn't do the same here for `Vector64<long>`

Reply:
@tannergooding do you think we can just simplify the C# signature to do the same? The current shape seems to be too complex to achieve this.
Or you think we need to do this expansion for both SIMD16 or SIMD8

Either is fine for me

Reply:
I think if we're going to do the expansion for V128, we should also do it for V64.

I don't think we can "simplify" it since at best we'd end up with:
```
if ((typeof(T) == typeof(long)) || (typeof(T) == typeof(ulong))
{
    return Create(left.AsUInt64().ToScalar(), right.AsUInt64().ToScalar());
}
else
{
    // for loop calling Scalar<T>.Multiply
}
```

The JIT will do the right thing here, but based on the recent conversations we had on wanting to keep "core" implementations in the JIT, it seems like its simpler to just handle them both explicitly (which avoids needing to inline and dead code eliminate the loop).

and notably the `for` loop itself is already "optimized" by the JIT today, since it's using `Count` as the upper limit, which means the JIT unrolls everything, that's part of why the perf gain even for V128 is so small (as we're already getting nearly optimal, with the few ns difference being minor things the JIT doesn't clean up today).

Reply:
For reference, `Vector64.Multiply` is generating this today. The only "suboptimal" bits is the zeroing of w0/w1, which I'm not quite sure why that's happening but it seems related to the loop (we don't get it for Vector64.Create(x.GetElement(0) * y.GetElement(0))`):
```asm
mov     w0, wzr
umov    x0, v0.d[0]
mov     w1, wzr
umov    x1, v1.d[0]
mul     x0, x0, x1
mov     v0.d[0], x0
```

For `Vector128.Multiply` we're generating a bit more, it again appears pessimized slightly from the overall loop, but also in part because we aren't optimizing `GetUpper().GetElement(#)` into just `GetElement(#)` like we could be:
```asm
mov     v16.16b, v0.16b
mov     v17.16b, v1.16b
mov     w0, wzr
umov    x0, v16.d[0]
mov     w1, wzr
umov    x1, v17.d[0]
mul     x0, x0, x1
mov     v16.d[0], x0
ext     v0.16b, v0.16b, v0.16b, #8
ext     v17.16b, v1.16b, v1.16b, #8
mov     w0, wzr
umov    x0, v0.d[0]
mov     w1, wzr
umov    x1, v17.d[0]
mul     x0, x0, x1
mov     v0.d[0], x0
ldr     q17, [fp, #0x10]	// [V15 tmp13]
ins     v17.d[0], v16.d[0]
ins     v17.d[1], v0.d[0]
mov     v0.16b, v17.16b
```

The more optimal sequence would be this, which is essentially what we get with your PR here (the `mov w0, #1` seems bogus here as well, maybe we're missing a containment opportunity or a dead use?):
```asm
umov    x0, v0.d[0]
umov    x1, v1.d[0]
mul     x0, x0, x1
ins     v16.d[0], x0
mov     w0, #1
umov    x0, v0.d[1]
mov     w1, #1
umov    x1, v1.d[1]
mul     x0, x0, x1
ins     v16.d[1], x0
mov     v0.16b, v16.16b
```

Reply:
ok, yeah, we're missing a containment opportunity on `GetElement`, that should be an easy thing to fix:
```
Generating: N019 (  1,  2) [000008] -----------                    t8 =    CNS_INT   int    1 REG x0 $41
IN0005:             mov     w0, #1
                                                                        /--*  t7     simd16 
                                                                        +--*  t8     int    
Generating: N021 (  3,  4) [000009] -----------                    t9 = *  HWINTRINSIC long   long GetElement REG x0 $1c0
```

Recognizing `GetUpper().GetElement(x)` and `GetLower().GetElement(x)` should also be something that's easy to do, I can get to them both after this PR goes in

Reply:
Yes, I noticed that too, even wanted to file an issue that 1 is not contained for GetElement